lib-src/FileDialog/configure.ac

```diff
@@ -122,6 +122,16 @@ AM_CONDITIONAL([GTK], test "$IMPLEMENTATION" = "gtk")
 AM_CONDITIONAL([MAC], test "$IMPLEMENTATION" = "mac")
 AM_CONDITIONAL([WINDOWS], test "$IMPLEMENTATION" = "win")
 
+dnl Automake uses the Ojective-C++ linker in any case, so we have to overwrite
+dnl the variables it's using with the correct C++ counterparts (only those used
+dnl for linking).
+if [[ "$IMPLEMENTATION" = gtk ]]
```

Comment:
It is inside if gtk, because in this case it is building for Linux.

The build command uses the objective C variables, that is to keep working on Mac. So in case it builds for Linux, these variables are filled with the content of the C++ variables. So the compiler does not need to support objective C. It uses the C++ compiler with the C++ compile flags.

But I have not tested with any other compiler than gcc. And I am by no means an experienced user of autotools.

If it is not necessary to support Mac with autotools anymore, the objective C parts should be removed completely in a separate PR.